src/common/reduction/Cluster.cpp

```diff
@@ -196,27 +181,46 @@ double Cluster::time_center2() const {
 double Cluster::coord_utpc(bool weighted) const {
   if (!weighted)
   {
-    return coord_utpc_/cnt_utpc_;
+    return coord_utpc_;
+  }
+
+  double n1 = 0, n2 = 0, n3 = 0, w1 = 0, w2 = 0, w3 = 0;
```

Comment:
You could make a case for using actual (and meaningful) names here :-)

Reply:
The intention was: n = numerator, w = weight
The numerator is either the strip or the time, and the weight is the charge or adc  value.